examples/recommendation/movie_recommendation.py

```diff
@@ -0,0 +1,145 @@
+from sklearn.metrics.pairwise import cosine_similarity
+
+#################################################################
+# Recommendation Engines
+#-----------------------
+# A Recommendation Engine seeks to predict the rating a user would
+# give to an item or to predict the item in order to meet the
+# user's taste. The more precise Recommendation Engines are, the
+# better they can guess a user's or customer's taste.
+# Recommendation Engines are used across various scientific and
+# economic fields, e.g. in playlist generators, product recommen-
+# ders in online stores or content recommenders on social media
+# platforms.
+#
+# Although all Recommendation Engines aim to achieve the same goal,
+# they use different techniques. The three main techniques are:
+# - Popularity-based
+# - Content-based
+# - Collaborative Filtering
+#
+# Popularity-based
+#-----------------
+# Popularity-based Recommendation Engines are the simplest form of
+# Recommendation Engine. They count the number of views or clicks
+# a product or service got and rank their popularity alongside
+# other products and services. The user then gets recommended the
+# currently most popular item(s). An example for Popularity-based
+# Recommendation Engines is YouTube's list of trending videos.
+#
+# Content-based
+#--------------
+# Content-based Recommendation Engines focus on the preferences
+# of the user currently using the service. They take into con-
+# sideration what content the user likes, then make out the key
+# components or features of the product or service and finally
+# search for other products and services with similar characte-
+# ristics to be recommended to that user.
+#
+# Collaborative Filtering
+#------------------------
+# Collabarative Filtering tries to identify similar types of
+# users based on their individual behavior and preferences.
+# If two users were clustered as members of a group of similar
+# users, the Recommendation Engine can suggest content that one
+# group member liked, to the other group members. Since they
+# have shown a related consumption behavior befor, chances are
+# hight that they will like the recommendation, too.
+
+#################################################################
+# Content-based Movie Recommendation Engine
+#------------------------------------------
+# In this example, our Recommendation Engine will be Content-based.
+# Consequently, we need to find similar movies to a given movie
+# in order to make a movie proposal the user might like.
+
+# Fetch dataset from OpenML and load it as a dataframe
+X, y = fetch_openml('movies', version=1, as_frame=True, return_X_y=True)
+X = X.astype({"index": int})
+
+# Choose relevant columns of the dataframe as feature set
+features = ["keywords", "cast", "genres", "director"]
+
+# Concatenating all extracted features into single string
+def concat_features(row):
+	return (row["keywords"] + " " + row["cast"] + " " +
```

Comment:
```suggestion
    return (row["keywords"] + " " + row["cast"] + " " +
```
You used a tab instead of 4 spaces. Python world does not use tabs.
Same happens a few times below.